src/Shared/ChangeWaves.cs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Shared;
+using System;
+using System.Linq;
+
+namespace Microsoft.Build.Utilities
+{
+    internal enum ChangeWaveConversionState
+    {
+        NotConvertedYet,
+        Valid,
+        InvalidFormat,
+        OutOfRotation,
+        InvalidVersion
+    }
+
+    /// <summary>
+    /// All waves are enabled by default, meaning all features behind change wave versions are enabled.
+    /// </summary>
+    public class ChangeWaves
+    {
+        public static readonly string[] AllWaves = { Wave16_8, Wave16_10, Wave17_0 };
+        public static readonly Version[] AllWavesAsVersion = Array.ConvertAll<string, Version>(AllWaves, Version.Parse);
+        public const string Wave16_8 = "16.8";
+        public const string Wave16_10 = "16.10";
```

Comment:
Maybe convert these to versions instead of strings? Assuming they average being used more than once, it's worth it.

Reply:
Worth doing, will service it with that in mind.